app/controllers/have_read_books_controller.rb

```diff
@@ -0,0 +1,19 @@
+class HaveReadBooksController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    user_book_ids = HaveReadBook.where(user: current_user).order(date_completed: :desc).pluck(:book_id)
+
+    if sort_params.present?
```

Comment:
You don't need to change it this way, but this should work as follows:
```@have_read_books = Book.where(id: user_book_ids)
@have_read_books = @have_read_books.order_by(sort_params) if sort_params.present?```

That said, you should be able to set up the associations to call `user.books` and get back all has_read_books.